sound/soc/sof/ipc.c

```diff
@@ -639,90 +639,75 @@ EXPORT_SYMBOL(snd_sof_ipc_stream_posn);
 /*
  * IPC get()/set() for kcontrols.
  */
-
-int snd_sof_ipc_set_comp_data(struct snd_sof_ipc *ipc,
-			      struct snd_sof_control *scontrol, u32 ipc_cmd,
-			      enum sof_ipc_ctrl_type ctrl_type,
-			      enum sof_ipc_ctrl_cmd ctrl_cmd)
+int snd_sof_ipc_set_get_comp_data(struct snd_sof_ipc *ipc,
+				  struct snd_sof_control *scontrol,
+				  u32 ipc_cmd,
+				  enum sof_ipc_ctrl_type ctrl_type,
+				  enum sof_ipc_ctrl_cmd ctrl_cmd)
 {
 	struct snd_sof_dev *sdev = ipc->sdev;
 	struct sof_ipc_ctrl_data *cdata = scontrol->control_data;
-	int err;
+	u32 send_bytes;
+	int send;
+	int err = 0;
 
-	/* read firmware volume */
+	/* read or write firmware volume */
 	if (scontrol->readback_offset != 0) {
```

Comment:
I think everything under this "if ()" looks like a good candidate for a new function